src/model/eraser/EraseableStroke.cpp

```diff
@@ -243,88 +231,75 @@ auto EraseableStroke::erasePart(double x, double y, double halfEraserSize, Erase
-
-    if (current) {
-        lists = g_list_append(lists, current);
-        current = nullptr;
+    if (splitPoints.back().empty()) {
+        splitPoints.pop_back();
     }
 
-    g_list_free(part->points);
-    part->points = nullptr;
-    if (lists) {
-        part->points = static_cast<GList*>(lists->data);
-        lists = g_list_delete_link(lists, lists);
+    points.clear();
+    if (!splitPoints.empty()) {
+        points = std::move(splitPoints.front());
+        splitPoints.erase(splitPoints.begin());
 
-        int pos = g_list_index(list->data, part) + 1;
+        PartList::iterator insertPos = partIter + 1;
 
         // create data structure for all new (splitted) parts
-        for (GList* l = lists; l != nullptr; l = l->next) {
-            auto* newPart = new EraseableStrokePart(part->width);
-            newPart->points = static_cast<GList*>(l->data);
-            list->data = g_list_insert(list->data, newPart, pos++);
+        for (auto& l: splitPoints) {
+            PartList::iterator newPart = list.emplace(insertPos, partIter->getWidth());
+            newPart->getPoints() = std::move(l);
+            insertPos++;
         }
-
-        g_list_free(lists);
     } else {
         // no parts, all deleted
-        list->data = g_list_remove(list->data, part);
         *deleteStrokeAfter = true;
     }
 
     return changed;
 }
 
-auto EraseableStroke::getStroke(Stroke* original) -> GList* {
-    GList* list = nullptr;
+auto EraseableStroke::getStroke(Stroke* original) -> std::vector<std::unique_ptr<Stroke>> {
+    std::vector<std::unique_ptr<Stroke>> strokeList;
 
-    Stroke* s = nullptr;
     Point lastPoint(NAN, NAN);
-    for (GList* l = this->parts->data; l != nullptr; l = l->next) {
-        auto* p = static_cast<EraseableStrokePart*>(l->data);
-        GList* points = p->getPoints();
-        if (g_list_length(points) < 2) {
+    for (EraseableStrokePart& part: parts) {
+        std::vector<Point>& points = part.getPoints();
+        if (points.size() < 2) {
             continue;
         }
 
-        Point a = *(static_cast<Point*>(g_list_first(points)->data));
-        Point b = *(static_cast<Point*>(g_list_last(points)->data));
-        a.z = p->width;
+        Point a = points.front();
+        Point b = points.back();
+        a.z = part.getWidth();
 
-        if (!lastPoint.equalsPos(a) || s == nullptr) {
-            if (s) {
-                s->addPoint(lastPoint);
+        if (!lastPoint.equalsPos(a) || strokeList.empty()) {
+            if (!strokeList.empty()) {
+                strokeList.back()->addPoint(lastPoint);
             }
-            s = new Stroke();
-            s->setColor(original->getColor());
-            s->setToolType(original->getToolType());
-            s->setLineStyle(original->getLineStyle());
-            s->setWidth(original->getWidth());
-            list = g_list_append(list, s);
+            auto& newStroke = strokeList.emplace_back(std::move(std::make_unique<Stroke>()));
```

Comment:
`std::make_unique<Stroke>()` is already a rvalue-reference, no need to call move here.
```suggestion
            auto& newStroke = strokeList.emplace_back(std::make_unique<Stroke>());
```